### ISSUE TITLE
content-type header missing on Android

### ISSUE THREAD

**Issue by isaachinman** [Vydia/react-native-background-upload#69]
Not sure if this is specific to an Android version, but on a OnePlus 3 running Android 8.0.0, the upload request hits my API without a `content-type` header set. The header is set correctly on iOS.

As a workaround, I've explicitly set the header in my options object:

    const options = {
      url: 'https://myapi/add-photo',
      path: image.uri,
      method: 'POST',
      type: 'raw',
      headers: {
        'content-type': 'application/octet-stream',
      },
      // Android-only options
      notification: {
        enabled: true,
      },
    }

Let me know if you need more info to debug.

**Comment by StevePotter** [comment]
@isaachinman how did you determine the header wasn't set?  Is that from the server side?  

Do you know how to debug the Java code?  If so, I'd like to see a breakpoint set at: https://github.com/Vydia/react-native-background-upload/blob/master/android/src/main/java/com/vydia/UploaderModule.java#L236

If possible, can you debug that and let me know what's going on? 

Thanks for the string of good issues!

**Comment by isaachinman** [comment]
@StevePotter Yes, I have header validation on my API so it was clear immediately. Unfortunately no, I'm not a Java dev, although at first glance I don't think that's the piece of code we're after. I was under the impression that `'content-type': 'application/octet-stream'` should be set regardless of any header options passed, for every single upload by default.

**Comment by TSMMark** [comment]
It looks like we do get a mimetype and add to headers for iOS here https://github.com/Vydia/react-native-background-upload/blob/c308eef5868c5a59d7680168eeace25aeedcdbb9/ios/VydiaRNFileUploader.m#L260

Perhaps something that should be added for android as well

**Comment by isaachinman** [comment]
Yes, headers are set correctly on iOS as per my original comment. Uniformity across both platforms reduces complexity for users, so this should definitely be implemented on Android, I just don't know how much help I can be.

**Comment by TSMMark** [comment]
Thanks for reporting this @isaachinman. In the meantime, I added content-type to the headers in the readme. https://github.com/Vydia/react-native-background-upload#usage

**Comment by StevePotter** [comment]
@isaachinman @TSMMark is right, we are setting content-type explicitly in iOS, but that's only for multipart file uploads, which is not the typical type of upload.  So by default, no headers get set at all.
